tests/rptest/tests/datalake/datalake_services.py

```diff
@@ -143,6 +144,32 @@ def table_created():
             f"Timed out waiting {namespace}.{table} to be created in the catalog"
         )
 
+    def get_max_translated_offset(self,
+                                  topic,
+                                  partition=0,
+                                  timeout=30,
+                                  backoff_sec=5):
+        self.wait_for_iceberg_table("redpanda", topic, timeout, backoff_sec)
+
+        def get_offsets():
+            try:
+                offsets = dict(
+                    map(
+                        lambda e: (e.engine_name(),
+                                   e.max_translated_offset(
+                                       "redpanda", topic, partition)),
+                        self.query_engines))
+                return True, offsets
+            except:
```

Comment:
qq: isn't this already handled in L187 below? I thought the following evaluates to false in None case and is retried (atleast that was the intention unless I goofed up somehting)

```
all([
                offset and offset <= max_offset
                for _, max_offset in offsets.items()
            ])
```

Reply:
oh wait I guess the condition should be offset >= max_offset, thats probably the issue?